lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -66,6 +68,23 @@ def object? object
       object.type.include? RDFVocabularies::PCDMTerms.Object
     end
 
+    # TODO: This is ugly. Are there clever ways to assemble a list of ancestors for a 
+    # given collection?
+    def collection_ancestor? collections
+      next_batch = []
+      current_collections = collections
+      while !current_collections.empty? do
```

Comment:
I'm a bit confused here.  what cause current collections to get empty? I don't see anything removing members from there.